pkg/azuredisk/azuredisk.go

```diff
@@ -398,3 +395,18 @@ func getSourceVolumeId(snapshot *compute.Snapshot) string {
 	}
 	return ""
 }
+
+func getValidCreationData(subscriptionID, resourceGroup, sourceResourceID string) compute.CreationData {
+	if sourceResourceID == "" {
+		return compute.CreationData{
+			CreateOption: compute.Empty,
+		}
+	}
+	if match := diskSnapshotPathRE.FindString(sourceResourceID); match == "" {
+		sourceResourceID = "/subscriptions/" + subscriptionID + "/resourceGroups/" + resourceGroup + "/providers/Microsoft.Compute/snapshots/" + sourceResourceID
```

Comment:
I think you may also add basic verification on `sourceResourceID`, e.g. it is only a snapshot name, and if not, return error.
And also may use a diskSnapshotPath const, and use fmt.SPrintf:
```
diskSnapshotPath = `"/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/snapshots/%s" sourceResourceID`
```

Reply:
done.
Here if the provided sourceResourceID is wrong, it will also be jointed. So we just need to judge it according to the amount of '/'. Have a look at some new mistake samples in the unit test.